pages/submissionFile/submissionFile.js

```diff
@@ -0,0 +1,70 @@
+// @ts-check
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const { isBinaryFile } = require('isbinaryfile');
+const mime = require('mime');
+const sqldb = require('@prairielearn/postgres');
+
+const sql = sqldb.loadSqlEquiv(__filename);
+const router = express.Router({ mergeParams: true });
+
+const MEDIA_PREFIXES = ['image/', 'audio/', 'video/', 'application/pdf'];
+
+/**
+ * Guesses the mime type for a file based on its name and contents.
+ *
+ * @param {string} name The file's name
+ * @param {Buffer} buffer The file's contents
+ * @returns {Promise<string>} The guessed mime type
+ */
+async function guessMimeType(name, buffer) {
+  const mimeType = mime.getType(name);
+  const isBinary = await isBinaryFile(buffer);
+
+  if (!isBinary) {
+    return 'text/plain';
```

Comment:
Is this actually the right thing to do here? For example, a `.svg` file would detect as `text/plain` with this check, I think?

No need to hold up this this PR over this concern -- we can always address this later.

Reply:
Good comment! I think I should actually swap the order here. That is, first check if it's a "media" mime (which would catch `image/svg+xml`), and only if it's not check if it's binary or not.

Reply:
Yep, agreed.

Reply:
Changed in c47f83b.